bundles/org.openhab.binding.intesis/src/main/java/org/openhab/binding/intesis/internal/handler/IntesisHomeHandler.java

```diff
@@ -0,0 +1,529 @@
+                        case "AUTO":
+                            value = 0;
+                            break;
+                        case "1":
+                        case "2":
+                        case "3":
+                        case "4":
+                        case "5":
+                        case "6":
+                        case "7":
+                        case "8":
+                        case "9":
+                            value = Integer.parseInt(command.toString());
+                            break;
+                        case "SWING":
+                            value = 10;
+                            break;
+                        case "SWIRL":
+                            value = 11;
+                            break;
+                        case "WIDE":
+                            value = 12;
+                            break;
+                    }
+                    switch (channelId) {
+                        case CHANNEL_TYPE_VANESUD:
+                            uid = 5;
+                            break;
+                        case CHANNEL_TYPE_VANESLR:
+                            uid = 6;
+                            break;
+                    }
+                    break;
+                case CHANNEL_TYPE_TARGETTEMP:
+                    uid = 9;
+                    if (command instanceof QuantityType) {
+                        QuantityType<?> newVal = (QuantityType<?>) command;
+                        newVal = newVal.toUnit(SIUnits.CELSIUS);
+                        if (newVal != null) {
+                            value = newVal.intValue() * 10;
+                        }
+                    }
+                    break;
+            }
+        }
+        if (uid != 0) {
+            login();
+            String contentString = "{\"command\":\"setdatapointvalue\",\"data\":{\"sessionID\":\"" + sessionId
+                    + "\", \"uid\":" + uid + ",\"value\":" + value + "}}";
+            String response = api.postRequest(ipAddress, contentString);
+            if (response != null) {
+                try {
+                    Response resp = gson.fromJson(response, Response.class);
+                    boolean success = resp.success;
+                    if (!success) {
+                        updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR);
+                        String sessionString = "{\"command\":\"login\",\"data\":{\"username\":\"Admin\",\"password\":\""
+                                + password + "\"}}";
+                        response = api.postRequest(ipAddress, sessionString);
+                    } else {
+                        logout();
+                        updateStatus(ThingStatus.ONLINE);
+                    }
+                } catch (JsonSyntaxException e) {
+                    updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR);
+                }
+            }
+        }
```

Comment:
`handleCommand` should execute and return quickly. Http requests can turn into lengthy operations so it would better if you ran this code asynchronously on the `scheduler`.

Reply:
Done